src/main/java/com/klodnicki/Bike/v2/model/entity/ChargingStation.java

```diff
@@ -39,4 +40,31 @@ public ChargingStation(Long id, String name, String address, String city, int fr
         this.freeSlots = freeSlots;
         this.bikeList = bikeList;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o != null && getClass() == o.getClass()) {
+            ChargingStation that = (ChargingStation) o;
+            if (freeSlots == that.freeSlots &&
+                    Objects.equals(id, that.id) &&
+                    Objects.equals(name, that.name) &&
+                    Objects.equals(address, that.address) &&
+                    Objects.equals(city, that.city) &&
+                    Objects.equals(rent, that.rent)) {
+
+                // Compare bikeList as sets
+                Set<Bike> thisBikeSet = bikeList == null ? new HashSet<>() : new HashSet<>(bikeList);
+                Set<Bike> thatBikeSet = that.bikeList == null ? new HashSet<>() : new HashSet<>(that.bikeList);
+                return thisBikeSet.equals(thatBikeSet);
+            }
```

Comment:
I'm not a fan of that nested layer of code as consequence of this rather big if.
I'd like to grasp a concept of changing the bikeList from List to HashSet. Could you explain why you chose this solution?

Reply:
yes, in Set the order of bikes is not important, while in List it is. Therefore if I have to bikeList with objects and I compare them to each other I don't have equality because order of bikes in each list can be different.